Extension/browser/safari/lib/converter.js

```diff
@@ -442,7 +442,29 @@ var SafariContentBlockerConverter = {
      */
     convertLine: function (ruleText, errors) {
         try {
-            if (ruleText === null || ruleText === '' || 
+            return this._convertAGRule(this.parseAGRule(ruleText, errors));
+        } catch (ex) {
+            var message = 'Error converting rule from: ' + ruleText + ' cause:\n' + ex;
+            message = ruleText + '\r\n' + message + '\r\n';
+            adguard.console.debug(message);
+
+            if (errors) {
+                errors.push(message);
+            }
+
+            return null;
+        }
+    },
+
+    /**
+     * Creates AG rule form text
+     *
+     * @param ruleText
+     * @param errors
+     */
+    parseAGRule: function (ruleText, errors) {
+        try {
+            if (ruleText === null || ruleText === '' ||
                 ruleText.indexOf('!') === 0 || ruleText.indexOf(' ') === 0 ||
                 ruleText.indexOf(' - ') > 0) {
```

Comment:
I don't quite understand why do you miss the rule that starts with space or contains `-`?